login-workflow/src/new-architecture/screens/AccountDetailsScreen/AccountDetailsScreenBase.tsx

```diff
@@ -0,0 +1,115 @@
+    const {
+        firstNameLabel,
+        initialFirstName,
+        firstNameValidator = (): boolean => true,
+        firstNameTextFieldProps,
+        lastNameLabel,
+        initialLastName,
+        lastNameValidator = (): boolean => true,
+        lastNameTextFieldProps,
+    } = props;
+
+    const cardBaseProps = props.WorkflowCardBaseProps || {};
+    const headerProps = props.WorkflowCardHeaderProps || {};
+    const instructionsProps = props.WorkflowCardInstructionProps || {};
+    const actionsProps = props.WorkflowCardActionsProps || {};
+
+    const firstRef = useRef<any>(null);
+    const lastRef = useRef<any>(null);
+
+    const [firstNameInput, setFirstNameInput] = React.useState(initialFirstName ? initialFirstName : '');
+    const [lastNameInput, setLastNameInput] = React.useState(initialLastName ? initialLastName : '');
+
+    const [firstNameError, setFirstNameError] = React.useState('');
+    const [lastNameError, setLastNameError] = React.useState('');
+
+    const [showFirstNameError, setShowFirstNameError] = React.useState(false);
+    const [showLastNameError, setShowLastNameError] = React.useState(false);
+
+    const [isFirstNameValid, setIsFirstNameValid] = React.useState(false);
+    const [isLastNameValid, setIsLastNameValid] = React.useState(false);
+    const [isFormValid, setIsFormValid] = React.useState(false);
+
+    const onFirstNameChange = useCallback(
+        (firstName: string) => {
+            setFirstNameInput(firstName);
+            const validatorResponse = firstNameValidator(firstName);
+
+            setIsFirstNameValid(typeof validatorResponse === 'boolean' ? validatorResponse : false);
+            setShowFirstNameError(typeof validatorResponse === 'boolean' ? false : true);
+            setFirstNameError(typeof validatorResponse === 'string' ? validatorResponse : '');
+        },
+        [firstNameValidator]
+    );
+
+    const onLastNameChange = useCallback(
+        (lastName: string) => {
+            setLastNameInput(lastName);
+            const validatorResponse = lastNameValidator(lastName);
+
+            setIsLastNameValid(typeof validatorResponse === 'boolean' ? validatorResponse : false);
+            setShowLastNameError(typeof validatorResponse === 'boolean' ? false : true);
+            setLastNameError(typeof validatorResponse === 'string' ? validatorResponse : '');
+        },
+        [lastNameValidator]
+    );
+
+    useEffect(() => {
+        setIsFormValid(isFirstNameValid && isLastNameValid);
+    }, [isFirstNameValid, isLastNameValid]);
+
+    return (
+        <WorkflowCard {...cardBaseProps}>
+            <WorkflowCardHeader {...headerProps} />
+            <WorkflowCardBody>
+                <WorkflowCardInstructions {...instructionsProps} divider />
+                <TextField
+                    id="first"
+                    fullWidth
+                    variant="filled"
+                    {...firstNameTextFieldProps}
```

Comment:
@joebochill @manojleaton Are we expecting every input to have these now?

Reply:
yes